File: app/src/main/java/com/chrisprime/primestationonecontrol/utilities/CommandUtilities.java
package com.chrisprime.primestationonecontrol.utilities;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.List;

import timber.log.Timber;

/**
 * Created by cpaian on 7/20/15.
 */
public class CommandUtilities {

    public static String doCommand(List<String> command)
    {
        String s = null;
        String commandResult = "";
        ProcessBuilder pb = new ProcessBuilder(command);
        try {
            Process process = pb.start();

            BufferedReader stdInput = new BufferedReader(new InputStreamReader(process.getInputStream()));
            BufferedReader stdError = new BufferedReader(new InputStreamReader(process.getErrorStream()));

            StringBuilder stringBuilder = new StringBuilder();

            // read the output from the command
            String stdOutTitle = "Here is the standard output of the command:\n";
            Timber.d(stdOutTitle);
            stringBuilder.append(stdOutTitle);
            while ((s = stdInput.readLine()) != null) {
                Timber.d(s);
                stringBuilder.append(s);
            }

            // read any errors from the attempted command
            String stdErrTitle = "Here is the standard error of the command (if any):\n";
            Timber.d(stdErrTitle);
            stringBuilder.append(stdErrTitle);
            while ((s = stdError.readLine()) != null) {
                Timber.d(s);
                stringBuilder.append(s);
            }
            commandResult = stringBuilder.toString();
            process.exitValue()
        } catch(IOException e) {
            Timber.e(e, "Exception executing command " + command);
        }
        return commandResult;
    }
}

File: app/src/main/java/com/chrisprime/primestationonecontrol/utilities/CommandUtilities.kt
package com.chrisprime.primestationonecontrol.utilities

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.util.ArrayList

import timber.log.Timber

/**
 * Created by cpaian on 7/20/15.
 */
object CommandUtilities {

    fun doCommand(command: List<String>): Int {
        var s: String? = null
        var processExitValue = 1   //Default to "errored"
        val processBuilder = ProcessBuilder(command)
        try {
            val process = processBuilder.start()

            val stdInput = BufferedReader(InputStreamReader(process.inputStream))
            val stdError = BufferedReader(InputStreamReader(process.errorStream))

            // read the output from the command
            Timber.d("Here is the standard output of the command:\n")
            s = stdInput.readLine()
            while (s != null) {
                Timber.d(s)
                s = stdInput.readLine()
            }

            // read any errors from the attempted command
            Timber.d("Here is the standard error of the command (if any):\n")
            s = stdError.readLine()
            while (s != null) {
                Timber.d(s)
                s = stdError.readLine()
            }

            processExitValue = process.waitFor()
        } catch (e: IOException) {
            Timber.e(e, "Exception executing command " + command)
        } catch (e: InterruptedException) {
            Timber.e(e, "Exception executing command, interrupted... " + command)
        }

        return processExitValue
    }
}
